Automattic/woocommerce-payments
# Extend create intentions endpoint

Fixes #4196 

#### Changes proposed in this Pull Request
Implement support for payment_method_types, metadata, customer and capture_method parameters.










#### Testing instructions





* create an order
* make an api call to `payments/orders/<order_id>/create_terminal_intent` endpoint with `payment_methods`, `metadata` and `customer_id` fields
* check the new created payment intent . The payment methods, metadata and customer should have been updated.
<img width="945" alt="Screenshot 2022-06-16 at 13 05 16" src="https://user-images.githubusercontent.com/82826872/174047094-fa2dec72-6132-4a48-959d-3cef685f2978.png">


-------------------


**Post merge**
